java/hiro-client/src/main/java/co/arago/hiro/client/rest/DefaultWebSocketClient.java

```diff
@@ -40,18 +42,145 @@
+    private AsyncHttpClient client;
     private final TokenProvider tokenProvider;
     private final int timeout;
     private final WebsocketType type;
     private final String urlParameters;
     private final WebSocketListener handler;
 
+    private final Map<String, Map> eventFilterMessages = new LinkedHashMap<>();
+
+    private class DefaultWebSocketListener implements WebSocketListener {
+
+        private boolean reconnect;
+
+        public DefaultWebSocketListener(boolean reconnect) {
+            this.reconnect = reconnect;
+        }
+
+        /**
+         * Invoked when the {@link WebSocket} is open.
+         *
+         * @param websocket
+         *            the WebSocket
+         */
+        @Override
+        public void onOpen(WebSocket websocket) {
+            final Map m = HiroCollections.newMap();
+            m.put("type", "open");
+            m.put("open", websocket.isOpen());
+
+            if (LOG.isLoggable(Level.FINEST)) {
+                LOG.log(Level.FINEST, "connected " + this);
+            }
+
+            if (handler != null) {
+                handler.onOpen(websocket);
+            }
+
+            process(logListener, JSONValue.toJSONString(m));
+            reconnect = false;
+        }
+
+        /**
+         * Invoked when the {@link WebSocket} is closed.
+         *
+         * @param websocket
+         *            the WebSocket
+         * @param code
+         *            the status code
+         * @param reason
+         *            the reason message
+         * 
+         * @see "http://tools.ietf.org/html/rfc6455#section-5.5.1"
+         */
+        @Override
+        public void onClose(WebSocket websocket, int code, String reason) {
+            final Map m = HiroCollections.newMap();
+            m.put("type", "close");
+            m.put("code", code);
+            m.put("reason", reason);
+
+            if (LOG.isLoggable(Level.FINEST)) {
+                LOG.log(Level.FINEST, "received close " + this);
+            }
+
+            if (handler != null) {
+                handler.onClose(websocket, code, reason);
+            }
+
+            process(logListener, JSONValue.toJSONString(m));
+
+            if (!reconnect)
+                reconnect();
+
+        }
+
+        /**
+         * Invoked when the {@link WebSocket} crashes.
+         *
+         * @param t
+         *            a {@link Throwable}
+         */
+        @Override
+        public void onError(Throwable t) {
+            final Map m = HiroCollections.newMap();
+            m.put("type", "error");
+            m.put("message", t.getMessage());
+            m.put("stack", stacktrace(t));
+
+            if (LOG.isLoggable(Level.FINEST)) {
+                LOG.log(Level.FINEST, "received error " + this, t);
+            }
+
+            if (handler != null) {
+                handler.onError(t);
+            }
+
+            process(logListener, JSONValue.toJSONString(m));
+
+            if (!reconnect)
```

Comment:
this is at least confusing if not wrong reconnect then reconnect()

Reply:
Yeah. I should rename this - this confused me as well.